src/components/Button.js

```diff
@@ -1,9 +1,54 @@
-import React from "react";
+/** @jsx jsx */
+
+import { css, jsx } from "@emotion/core";
 
 export default function Button(props) {
+  const commonCss = `
+    border: transparent;
+    border-radius: 6px;
+    text-transform: capitalize;
+    -webkit-user-select: none;
+    -moz-user-select: none;
+    -ms-user-select: none;
+    user-select: none;
+    background: linear-gradient(to bottom,#8434d6 0%,#8434d6 100%);
+    color: #fff;
+    cursor: pointer;
+    margin: 8px;
+    &:hover {
+      background: linear-gradient(to bottom,#9e5edf 0%,#9e5edf 100%);
+    }
+  `
+
+  const cssOpts = {
+    medium: `
+      font-size: 18px;
+      font-weight: 500;
+      height: 48px;
+      padding: 10px 25px;
+    `,
+    large: `
+      font-size: 22px;
+      font-weight: 800;
+      height: 54px;
+      padding: 12px 27px;
+    `
+  }
+
```

Comment:
I've found that it's best practice to declare variables and functions that do not depend on props outside of the function. This prevents the component from becoming a bloated mess!